### PR TITLE
[Test] Reuse rest across test projects

### PR DESCRIPTION
**Pull request by dkijania**

### CODE REVIEW

File: testing/jormungandr-integration-tests/src/common/jormungandr/rest.rs [hunk 1]
```diff
@@ -10,18 +16,24 @@ pub enum RestError {
     CannotDeserialize(#[from] serde_json::Error),
     #[error("could not send reqeuest")]
     RequestError(#[from] reqwest::Error),
+    #[error("hash parse error")]
+    HashParseError(#[from] chain_crypto::hash::Error)
 }
 
 /// Specialized rest api
-#[derive(Debug)]
+#[derive(Debug,Clone)]
```
Comment (NicolasDP) [inline testing/jormungandr-integration-tests/src/common/jormungandr/rest.rs:24]: add some spacing in your derive please
```suggestion
#[derive(Debug, Clone)]
```

File: testing/jormungandr-integration-tests/src/common/jormungandr/mod.rs [hunk 2]
```diff
@@ -8,7 +8,7 @@ pub use benchmark::storage_loading_benchmark_from_log;
 pub use configuration_builder::ConfigurationBuilder;
 pub use logger::{JormungandrLogger, LogEntry};
 pub use process::*;
-pub use rest::JormungandrRest;
+pub use rest::{JormungandrRest,RestError};
```
Comment (NicolasDP) [inline testing/jormungandr-integration-tests/src/common/jormungandr/mod.rs:11]: aerate your imports, a space between `,` and the following character
```suggestion
pub use rest::{JormungandrRest, RestError};
```

File: jormungandr-lib/src/interfaces/peer_stats.rs [hunk 3]
```diff
@@ -29,12 +29,7 @@ pub struct Profile {
     pub subscriptions: Vec<Subscription>,
 }
 
-#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
-#[serde(deny_unknown_fields)]
-pub struct Info {
-    pub address: String,
-    pub id: String,
-}
+pub type Info = String;
```
Comment (mzabaluev) [inline jormungandr-lib/src/interfaces/peer_stats.rs:32]: If you want to keep the meaning as "peer info" I'd leave it as a one-field struct for future proofing.

File: testing/jormungandr-scenario-tests/src/node.rs [hunk 4]
```diff
@@ -7,7 +7,7 @@ use chain_impl_mockchain::{
 };
 use indicatif::ProgressBar;
 use jormungandr_integration_tests::{
-    common::jormungandr::logger::JormungandrLogger,
+    common::jormungandr::{logger::JormungandrLogger, JormungandrRest, RestError},
```
Comment (mzabaluev) [inline testing/jormungandr-scenario-tests/src/node.rs:10]: Shouldn't we move all reusable stuff to `jormungandr-testing-utils` to reduce the cross-crate dependency spaghetti?

Comment (NicolasDP) [reply testing/jormungandr-scenario-tests/src/node.rs:10]: agreed

Comment (dkijania) [reply testing/jormungandr-scenario-tests/src/node.rs:10]: also agreed, but test project refactoring is not a highest priority right now. I want to do that in many small steps. this is intermediate step which moves some of rest methods to integration test and then move it to testing-utils.

File: testing/jormungandr-integration-tests/src/common/jormungandr/mod.rs (hunk 2 above)
Comment (dkijania) [reply testing/jormungandr-integration-tests/src/common/jormungandr/mod.rs:11]: Apologies for that and other fmt issues. i was running `cargo fmt` in testing directory not in the root folder

**Pull request merged**